recipe/patches/25786.patch

```diff
@@ -0,0 +1,101 @@
+
+fixes introspection with ? and ??
+--- a/src/doc/en/introspect/__init__.py
++++ /dev/null
+@@ -1 +0,0 @@
+-
+--- a/src/doc/en/introspect/conf.py
++++ /dev/null
+@@ -1,20 +0,0 @@
+-# -*- coding: utf-8 -*-
+-# Sage introspection build configuration file.
+-# See sagenb.notebook.cell.Cell.set_introspect_html() for details.
+-
+-import sys, os
+-sys.path.append(os.environ['SAGE_DOC_SRC'])
+-from common.conf import *
+-
+-extensions = ['sphinx.ext.autodoc', 'sphinx.ext.mathjax', 'sphinx.ext.todo',
+-              'sphinx.ext.extlinks']
+-
+-templates_path = ['templates']
+-html_static_path = ['static']
+-
+-html_use_modindex = False
+-html_use_index = False
+-html_split_index = False
+-html_copy_source = False
+-
+-todo_include_todos = True
+--- a/src/doc/en/introspect/static/empty
++++ /dev/null
+@@ -1 +0,0 @@
+-
+--- a/src/doc/en/introspect/templates/layout.html
++++ /dev/null
+@@ -1,3 +0,0 @@
+-<div class="docstring">
+-    {% block body %} {% endblock %}
+-</div>
+--- a/src/sage/misc/sphinxify.py
++++ b/src/sage/misc/sphinxify.py
+@@ -22,7 +22,6 @@
+ import re
+ import shutil
+ from tempfile import mkdtemp
+-from sage.env import SAGE_DOC_SRC
+ from sphinx.application import Sphinx
+ 
+ 
+@@ -78,12 +77,36 @@
+ 
+     # Sphinx constructor: Sphinx(srcdir, confdir, outdir, doctreedir,
+     # buildername, confoverrides, status, warning, freshenv).
+-    confdir = os.path.join(SAGE_DOC_SRC, 'en', 'introspect')
++    confdir = os.path.join(srcdir, 'en' , 'introspect')
++    os.makedirs(confdir)
++    with open(os.path.join(confdir, 'conf.py'), 'w') as filed:
++        filed.write(r"""
++extensions = ['sphinx.ext.autodoc', 'sphinx.ext.mathjax', 'sphinx.ext.todo', 'sphinx.ext.extlinks']
++
++templates_path = ['templates']
++html_static_path = ['static']
++
++html_use_modindex = False
++html_use_index = False
++html_split_index = False
++html_copy_source = False
++
++todo_include_todos = True""")
++    templatesdir = os.path.join(confdir, 'templates')
++    os.makedirs(templatesdir)
++    with open(os.path.join(templatesdir, 'layout.html'), 'w') as filed:
++        filed.write(r"""
++<div class="docstring">
++    {% block body %} {% endblock %}
++</div>""")
++    staticdir = os.path.join(confdir, 'static')
++    os.makedirs(staticdir)
++    with open(os.path.join(staticdir, 'empty'), 'w') as filed: pass
+ 
+-    open(os.path.join(srcdir, 'docutils.conf'), 'w').write(r"""
++    with open(os.path.join(srcdir, 'docutils.conf'), 'w') as filed:
++        filed.write(r"""
+ [parsers]
+-smart_quotes = no
+-""")
++smart_quotes = no""")
+     doctreedir = os.path.join(srcdir, 'doctrees')
+     confoverrides = {'html_context': {}, 'master_doc': 'docstring'}
+ 
+--- a/src/sage_setup/docbuild/build_options.py
++++ b/src/sage_setup/docbuild/build_options.py
+@@ -9,7 +9,6 @@
+ LANGUAGES = [d for d in os.listdir(SAGE_DOC_SRC) if re.match('^[a-z][a-z]$', d)]
+ SPHINXOPTS = ""
+ PAPER = ""
+-OMIT = ["introspect"]  # docs/dirs to omit when listing and building 'all'
```

Comment:
@saraedum, this is used in `src/sage_setup/docbuild/__init__.py` and `src/sage/misc/sagedoc.py`